### ISSUE TITLE
Detailed help for default command is inaccessible

### ISSUE THREAD

**Issue by cspotcode** [arcanis/clipanion#95]
When running `clitool --help` with a default command and other subcommands, clipanion will output a list of commands.  It will say `You can also print more details about any of these commands by calling them with the -h,--help flag right after the command name.`

However, the default command does not have a command name, so there is no way to get more details about the default command.

I'm not sure if/how this should be supported.  If the default command is given an additional, non-default path, *and* if that non-default path is specified *before* `Command.Default`, then the path will appear in `--help` documentation.  The output will not, however, indicate that this command is the default.

**Comment by arcanis** [comment]
Did you check with 3.0.1? It should be fixed starting from 2e910239f692ff59c13efff260eebba119eef1cb

**Comment by arcanis** [comment]
Although this behavior only triggers if you don't have other documented commands. If you do, then your best option is to also give a name to your "main" command (just like we do with `yarn`, which also has its own name via `yarn install`).

**Comment by cspotcode** [comment]
Right, this issue is about when you have other documented commands.

**Comment by paul-soporan** [comment]
> so there is no way to get more details about the default command.

There actually is, but it's undocumented. `--help` can work as a bound option that accepts a number, causing `--help=<n>` to display the help of the nth command registration with the provided path.

In Yarn's case for example:
- running `yarn --help=0` or `yarn -h=0` displays the usage of the [`EntryCommand`](https://github.com/yarnpkg/berry/blob/master/packages/plugin-essentials/sources/commands/entries/run.ts):
- running `yarn --help=1` or `yarn -h=1` displays the usage of the [`InstallCommand`](https://github.com/yarnpkg/berry/blob/master/packages/plugin-essentials/sources/commands/install.ts)

Once I finish the completion PR, `--help=<tab>` and `-h=<tab>` will be automatically completed:

![image](https://user-images.githubusercontent.com/32596136/130447901-0ad79d93-4f6d-4014-b607-a946c8305ecf.png)
